src/client/components/pages/parts/add-to-collection-modal.js

```diff
@@ -0,0 +1,309 @@
+import * as bootstrap from 'react-bootstrap';
+import CustomInput from '../../../input';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactSelect from 'react-select';
+import SelectWrapper from '../../input/select-wrapper';
+import request from 'superagent';
+
+
+const {Alert, Col, Button, Modal} = bootstrap;
+
+class AddToCollectionModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			collectionsAvailable: [],
+			message: {
+				text: null,
+				type: null
+			},
+			selectedCollections: [],
+			show: this.props.show,
+			showCollectionForm: false
+		};
+
+		this.handleClose = this.handleClose.bind(this);
+		this.getCollections = this.getCollections.bind(this);
+		this.handleAddToCollection = this.handleAddToCollection.bind(this);
+		this.toggleRow = this.toggleRow.bind(this);
+		this.handleShowCollectionForm = this.handleShowCollectionForm.bind(this);
+		this.handleShowAllCollections = this.handleShowAllCollections.bind(this);
+		this.handleAddToNewCollection = this.handleAddToNewCollection.bind(this);
+		this.isValid = this.isValid.bind(this);
+	}
+
+	async componentDidUpdate(prevProps) {
+		const collections = await this.getCollections();
+		if (prevProps.show !== this.props.show) {
+			// eslint-disable-next-line react/no-did-update-set-state
```

Comment:
Not 100% sure what's going on here.
Why do you need to store `show` on the state, instead of using `this.props.show` directly?

As for setting collectionsAvailable, I think it would be better to do it in `componentDidMount` instead. I don't really think we need to re-fetch if a user closes and reopens the modal.

Reply:
Yeah we can use `componentDidMount`. I was using `componentDidUpdate` because earlier I didn't have the feature of creating the collection in the modal. and had to open a new tab to create a new collection. In that case `componentDidUpdate` was required.
But now that we can create collection in the modal itself we don't need to fetch collections everytime. One time is enough